### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -23,7 +23,7 @@ resource "google_storage_bucket" "bucket" {
       condition {
         age                        = try(lifecycle_rule.value.condition.age, null)
         created_before             = try(lifecycle_rule.value.condition.created_before, null)
-        with_state                 = try(lifecycle_rule.value.condition.with_state, null)
+        with_state                 = try(upper(lifecycle_rule.value.condition.with_state), null)
```
Comment (mariux) [inline main.tf:26]: we should not silently transform user input.... later state can be "LIVEv2" whatever and we will not be able to change this... better add validation of specific values.

**Review by mariux — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (soerenmartius) [reply main.tf:26]: ok, will add validation in a follow up task and get rid of this now

Comment (soerenmartius) [reply main.tf:26]: resolved

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,9 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+## [0.0.5]
+
+### Added
+
+- Add unit tests
+
 ### Fixed
 
-- Fix attributes in nested terraform-google-storage-bucket-iam to not fail when `module_enabled = false`
+- Fix attributes in nested `terraform-google-storage-bucket-iam` module to not fail when `module_enabled = false`
```
Comment (mariux) [inline CHANGELOG.md:18]: this fix does not seem to be included

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**